lib/server/connect.js

```diff
@@ -33,8 +16,55 @@ SubHandlers = {
   // count of remote admins
   listeners: []
 };
+
 var uniqueId = 0;
 var serverId = Random.id();
+var startTime = new Date();
+
+KdData = new KdDataLayer({serverId: serverId});
+
+var traceStore = new TraceStore({kdData: KdData});
+traceStore.start();
+
+// Observers: KdConfig/listenersCount
+KdData.configColl.find({_id: 'listenersCount'}).observeChanges({
+  added: function(id, doc) {
+    SubHandlers.listeners.forEach(function(sub) {
+      sub.added('kdInfo', 'listeners-count', {count: doc.count});
+    });
+  },
+
+  changed: function(id, doc) {
+    SubHandlers.listeners.forEach(function(sub) {
+      sub.changed('kdInfo', 'listeners-count', {count: doc.count});
+    });
+  }
+});
+
+// Observers: KdTimeEvents/serverTimeEvents
+KdData.timeEventsColl.find().observeChanges({
```

Comment:
Here add a query to get the data only greater than the current time.
Otherwise we'll send the whole data set in the collection.